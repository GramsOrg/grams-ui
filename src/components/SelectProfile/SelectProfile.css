.avatar {
    margin-bottom: 0.3em;
}

.centered {
    text-align: center;
    width: 100%;
    align-items: center;
}

.top {
    vertical-align: top;
}
