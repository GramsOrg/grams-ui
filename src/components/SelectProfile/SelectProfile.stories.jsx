import React from 'react';

import { action } from "@storybook/addon-actions";

import SelectProfile from './SelectProfile.tsx';

export default {
  title: 'Auth/SelectProfile',
  component: SelectProfile,
  parameters: {
    componentSubtitle: 'Displays a list of profiles with an option to create a new profile',
    storyshots: { disable: false },
  },
  argTypes: {
    horizontal: {
      control: 'boolean'
    }
  }
};

const Template = ({ ...props }) => {
  return (
    <SelectProfile
      {...props}
      onCreate={action("Trigger onCreate()")}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
}

const profiles = [
  {
    id: "first",
    name: "Elliot",
    avatarUrl: "https://react.semantic-ui.com/images/avatar/large/elliot.jpg"
  }, {
    id: "second",
    name: "Jenny",
    isDeveloper: true,
    avatarUrl: "https://react.semantic-ui.com/images/avatar/large/jenny.jpg"
  }
]

export const Populated = Template.bind({});
Populated.args = {
  profiles: profiles,
  onSelect: action("Trigger onSelect()")
}

export const Horizontal = Template.bind({});
Horizontal.args = {
  horizontal: true,
  profiles: profiles,
  onSelect: action("Trigger onSelect()")
}
