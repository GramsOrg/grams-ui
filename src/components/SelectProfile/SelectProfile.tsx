import React from 'react';

import { useTranslation } from 'react-i18next';

import { useDirection } from '../../';

import { UserProfile } from '../../types/UserProfile';

import { Label, List } from 'semantic-ui-react';
import Avatar from 'react-avatar';

import './SelectProfile.css';

export interface IProfileItemsProps {
  profiles?: UserProfile[];
  onSelect?: (profile: UserProfile) => void;
}

export interface ISelectProfileProps {

  /**
   * List of profiles that will be displayed
   */
  profiles?: UserProfile[];

  /**
   * Triggers on user clicking the create button
   */
  onCreate?: () => void;

  /**
   * Triggers on user clicking on a profile
   */
  onSelect?: (profile: UserProfile) => void;

  /**
   * Toggles between displaying the profiles horizontally or vertically
   */
  horizontal?: boolean;
}

const defaultProps = {
  horizontal: false
};

const SelectProfile = (props: ISelectProfileProps) => {

  const { t } = useTranslation();
  const direction = useDirection();

  return (
    <List
      horizontal={ props?.horizontal || false }
      relaxed
      size="big"
      className={`centered top`}
      dir={direction}
    >
      <ProfileItems
        profiles={props?.profiles}
        onSelect={props?.onSelect}
      />
      <List.Item onClick={() => props?.onCreate?.()}>
        <Avatar
          name="+"
          size="64"
          round
          color='teal'
          title={t('SelectProfile.new')}
          className="avatar"
        />
        <List.Header>{t('SelectProfile.new')}</List.Header>
      </List.Item>
    </List>
  );
};

const ProfileItems = (props: IProfileItemsProps) => {

  const { t } = useTranslation();

  return (
    <>
      {props?.profiles?.map(profile => (
        <List.Item
          className='top'
          key={profile.id}
          onClick={() => props?.onSelect?.(profile)}
        >
          <Avatar
            name={profile.name}
            size="64"
            round
            src={profile.avatarUrl}
            className="avatar"
          />
          <List.Header>{profile.name}</List.Header>
          <List.Content>
          {profile.isDeveloper && (
            <Label color="blue" size="mini">
              {t('SelectProfile.dev')}
            </Label>
          )}
          </List.Content>
        </List.Item>
      ))}
    </>
  )
}

SelectProfile.defaultProps = defaultProps;

export default SelectProfile;
