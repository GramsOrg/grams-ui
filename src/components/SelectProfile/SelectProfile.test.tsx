/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import SelectProfile from './SelectProfile';

describe('SelectProfile Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<SelectProfile />);
    expect(asFragment()).toMatchSnapshot();
  });
});
