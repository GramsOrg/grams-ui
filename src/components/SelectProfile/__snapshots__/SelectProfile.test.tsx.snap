// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectProfile Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui big relaxed list centered top"
    dir="ltr"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="avatar sb-avatar"
        style="display: inline-block; vertical-align: middle; width: 64px; height: 64px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
      >
        <div
          class="avatar sb-avatar__text"
          style="width: 64px; height: 64px; line-height: initial; text-align: center; color: rgb(255, 255, 255); background: teal; border-radius: 100%;"
          title="New"
        >
          <div
            style="display: table; table-layout: fixed; width: 100%; height: 100%;"
          >
            <span
              style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
            >
              <span />
            </span>
          </div>
        </div>
      </div>
      <div
        class="header"
      >
        New
      </div>
    </div>
  </div>
</DocumentFragment>
`;
